src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
+            except Exception as e:
+                self.logger.error(f"Failed to connect to schedd: {str(e)}")
+                raise
+            
+            # Query credential daemon
+            self.logger.debug(f"Querying for credential daemon on {self.config.submit_host}")
             cred_ads = self.collector.query(
                 htcondor.AdTypes.Credd,
-                constraint=f'Name == "{submit_host_url}"'
+                constraint=f'Name == "{self.config.submit_host}"'
             )
 
             if not cred_ads:
-                print(f"No credential daemon found for {submit_host_url}. Continuing without it.")
+                self.logger.warning(f"No credential daemon found for {self.config.submit_host}. Continuing without it.")
                 self.credd = None
             else:
                 cred_ad = cred_ads[0]
-                print(f"Found credential daemon: {cred_ad.get('Name', 'Unknown')}")
+                self.logger.debug(f"Found credential daemon: {cred_ad.get('Name', 'Unknown')}")
                 self.credd = htcondor.Credd(cred_ad)
                 
                 # Add credentials for required services
-                self._add_credentials()
+                if self.credd:
+                    for service in ["rdrive", "scitokens"]:
+                        try:
+                            self.credd.add_user_service_cred(htcondor.CredType.OAuth, b"", service)
+                            self.logger.debug(f"Added credential for service: {service}")
+                        except Exception as e:
+                            self.logger.warning(f"Could not add credential for {service}: {e}")
 
-            logger.info("Successfully connected to HTCondor submit node")
+            self.logger.info("Successfully connected to HTCondor submit node")
         except Exception as e:
-            logger.error(f"Failed to setup HTCondor connection: {e}")
+            self.logger.error(f"Failed to setup HTCondor connection: {e}")
             raise
 
-    def submit_jobs(self, submit_config: dict) -> int:
+    def submit_jobs(self, files_txt_path: str) -> int:
         """Submit jobs to HTCondor using files.txt"""
         try:
-            # Add credentials for required services
-            self._add_credentials()
-
-            # create the Submit object
-            submit_obj = htcondor.Submit(submit_config)
+            abs_files_txt = os.path.abspath(files_txt_path)
+            
+            # Read the files list
+            with open(abs_files_txt, 'r') as f:
+                files = [line.strip() for line in f if line.strip()]
+            
+            self.logger.debug(f"Submitting jobs for {len(files)} files")
 
-            # Submit the job
-            result = self.schedd.submit(submit_obj, spool=True)
-            cluster_id = result.cluster()
+            # Create submit description
+            submit_desc = htcondor.Submit({
```

Comment:
I don't love that this job submit config is hardcoded into the same function that submits the job. ideally I think this function should be for submitting a job, whatever the job may be. the job submit config should be defined upstream and then passed as a parameter into this function. (please review the code that this PR removes)